pkg/endpoint/manager.go

```diff
@@ -25,75 +25,39 @@ import (
 )
 
 type endpointManager interface {
-	AllocateID(id uint16) (uint16, error)
-	RunK8sCiliumEndpointSync(*Endpoint, EndpointStatusConfiguration)
-	UpdateReferences(id.Identifiers, *Endpoint)
-	UpdateIDReference(*Endpoint)
 	RemoveReferences(id.Identifiers)
 	RemoveID(uint16)
 	ReleaseID(*Endpoint) error
-	AddIPv6Address(addressing.CiliumIPv6)
 	RemoveIPv6Address(addressing.CiliumIPv6)
 }
 
-// Expose exposes the endpoint to the endpointmanager. After this function
-// is called, the endpoint may be accessed by any lookup in the endpointmanager.
-func (e *Endpoint) Expose(mgr endpointManager) error {
+// Start assigns a Cilium Endpoint ID to the endpoint and prepares it to
+// receive events from other subsystems.
+//
+// The endpoint must not already be exposed via the endpointmanager prior to
+// calling Expose(), as it assumes unconditional access over the Endpoint
+// object.
```

Comment:
Is this comment still relevant?

Reply:
At the very least it needs some cleanup, since it refers to a function name that doesn't exist ;)